umm-spec-lib/test/cmr/umm_spec/test/validation/related_url.clj

```diff
@@ -19,60 +19,60 @@
                      {:RelatedUrls [{:URL "http://fresc.usgs.gov/products/dataset/moorhen_telemetry.zip"
                                      :URLContentType "DistributionURL"
                                      :Type "GET DATA"
-                                     :SubType "ECHO"}
+                                     :Subtype "ECHO"}
```

Comment:
Shouldn't the JSON schema validation have caught that we were trying to use SubType instead of Subtype for Related URLs in our tests and failed?

If not it seems like there might be a problem with the way RelatedUrls are defined in our JSON schema. @eereiter - can you take a look at that?

Reply:
```
    "RelatedUrlType": {
      "type": "object",
      "additionalProperties": false,
      "description": "Represents Internet sites that contain information related to the data, as well as related Internet sites such as project home pages, related data archives/servers, metadata extensions, online software packages, web mapping services, and calibration/validation data.",
      "properties": {
        "Description": {
          "description": "Description of the web page at this URL.",
          "type": "string",
          "minLength": 1,
          "maxLength": 4000
        },
        "URLContentType": {
            "description": "A keyword describing the distinct content type of the online resource to this resource. (e.g., 'DATACENTER URL', 'DATA CONTACT URL', 'DISTRIBUTION URL').",
            "$ref": "#/definitions/RelatedURLContentTypeEnum"
        },
        "Type": {
            "description": "A keyword describing the type of the online resource to this resource. This helps the GUI to know what to do with this resource. (e.g., 'GET DATA', 'GET SERVICE', 'GET VISUALIZATION').",
            "$ref": "#/definitions/RelatedUrlTypeEnum"
        },
        "Subtype": {
            "description": "A keyword describing the subtype of the online resource to this resource. This further helps the GUI to know what to do with this resource. (e.g., 'MEDIA', 'BROWSE', 'OPENDAP', 'OPENSEARCH', 'WEB COVERAGE SERVICES', 'WEB FEATURE SERVICES', 'WEB MAPPING SERVICES', 'SSW', 'ESI').",
            "$ref": "#/definitions/RelatedURLSubTypeEnum"
        },
```

Reply:
```
(defn- valid-url-content-types-combo?
  "Returns true if valid Type and SubType for URLContentType, nil otherwise."
  [url-content-type type sub-type]
  (when (and (some #(= type %) (su/valid-types-for-url-content-type url-content-type))
             (or (nil? sub-type)
                 (some #(= sub-type %) (su/valid-subtypes-for-type url-content-type type))))
    true))
```
sub-type was allowed to be nil in the combination check.

Reply:
I found the schema for RelatedUrls and it does have "additionalProperties": false. Maybe it is just treating fields case insensitively when performing validation? In any case no change needed to the schema.